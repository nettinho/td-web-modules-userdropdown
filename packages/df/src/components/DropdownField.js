import React from "react";
import PropTypes from "prop-types";
import { Form } from "semantic-ui-react";
import { injectIntl } from "react-intl";

export const DropdownField = ({
  intl: { formatMessage },
  label,
  name,
  required,
  values = [""],
  value,
  onChange,
  type
}) => (
  <Form.Field required={required}>
    <label>{label}</label>
    <Form.Dropdown
      fluid
      search
      selection
      placeholder={formatMessage({ id: "fields.dropdown.placeholder" })}
      onChange={onChange}
      name={name}
      value={value}
      multiple={type === "variable_list"}
      options={["", ...values].map((o, i) => ({
        key: i,
        value: o,
        text: o
      }))}
    />
  </Form.Field>
);

DropdownField.propTypes = {
  intl: PropTypes.object,
  label: PropTypes.string,
  name: PropTypes.string,
  required: PropTypes.bool,
  values: PropTypes.array,
  value: PropTypes.oneOfType([PropTypes.string, PropTypes.array]),
  onChange: PropTypes.func
};

export default injectIntl(DropdownField);
