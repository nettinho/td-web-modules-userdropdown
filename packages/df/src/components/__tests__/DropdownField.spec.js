import React from "react";
import { shallowWithIntl } from "intl-stub";
import { DropdownField } from "../DropdownField";

describe("<DropdownField />", () => {
  const label = "label";
  const name = "name";
  const required = true;
  const value = "Foo";
  const onChange = jest.fn();
  const props = { label, name, required, value, onChange };

  it("matches the latest snapshot", () => {
    const wrapper = shallowWithIntl(<DropdownField {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
