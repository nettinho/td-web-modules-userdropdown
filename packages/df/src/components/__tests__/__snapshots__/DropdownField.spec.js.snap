// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DropdownField /> matches the latest snapshot 1`] = `
<FormField
  required={true}
>
  <label>
    label
  </label>
  <FormDropdown
    as={[Function]}
    control={[Function]}
    fluid={true}
    multiple={false}
    name="name"
    onChange={[MockFunction]}
    options={
      Array [
        Object {
          "key": 0,
          "text": "",
          "value": "",
        },
        Object {
          "key": 1,
          "text": "",
          "value": "",
        },
      ]
    }
    placeholder="fields.dropdown.placeholder"
    search={true}
    selection={true}
    value="Foo"
  />
</FormField>
`;
